pkg/api/unversioned/helpers.go

```diff
@@ -181,3 +184,34 @@ func ExtractGroupVersions(l *APIGroupList) []string {
 	}
 	return groupVersions
 }
+
+// ExtractGroupVersionResourcesWithVerbs extracts the GroupVersionResources from a map. In case of
+// errors, an aggregate error is returned and those resources which could be parsed successfully.
+func ExtractGroupVersionResources(rls []*APIResourceList) ([]schema.GroupVersionResource, error) {
```

Comment:
And in discovery it could take the return value of whereever we get this to filtering it through as a one liner.